frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts

```diff
@@ -0,0 +1,103 @@
+import {
+	AttributeKeyOptions,
+	getAttributesKeys,
+	getAttributesValues,
+} from 'api/queryBuilder/getAttributesKeysValues';
+import { useRef, useState } from 'react';
+import { useQuery } from 'react-query';
+import { useDebounce } from 'react-use';
+import { IBuilderQueryForm } from 'types/api/queryBuilder/queryBuilderData';
+import { getCountOfSpace } from 'utils/getCountOfSpace';
+import { separateSearchValue } from 'utils/separateSearchValue';
+
+type UseFetchKeysAndValuesReturnT = {
+	keys: AttributeKeyOptions[];
+	results: string[];
+};
+
+export const useFetchKeysAndValues = (
+	searchValue: string,
+	query: IBuilderQueryForm,
+): UseFetchKeysAndValuesReturnT => {
+	const [keys, setKeys] = useState<AttributeKeyOptions[]>([]);
+	const [results, setResults] = useState<string[]>([]);
+
+	useQuery(
```

Comment:
This is a common hook to fetch the key + operator and value w.r.t to the key + operator based on the user search value or selected value.
It will return keys as selected key + operator and result as a derived value.

@palashgdev